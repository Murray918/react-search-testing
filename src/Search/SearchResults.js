import React from 'react'
import PropTypes from 'prop-types'

const SearchResults = ({ articles }) => {
	return (
		<ul className="search-results">
			{articles.map(({webUrl, webTitle}) => {
				return (
					<li key={webTitle}>
						<a href={webUrl} target="_blank" rel="noopener noreferrer">
							{webTitle}
						</a>
					</li>
				)
			})}
		</ul>
	)
}

export default SearchResults

SearchResults.propTypes = {
	articles: PropTypes.array
}
