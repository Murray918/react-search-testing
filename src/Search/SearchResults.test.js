import React from 'react'
import { shallow } from 'enzyme'
import SearchResults from './SearchResults'

const mockArticles = [
	{
		webUrl: 'www.google.com',
		webTitle: "Lets freaking party y'all!!!!"
	}
]

describe('Search Results', () => {
	test('renders', () => {
		const wrap = shallow(<SearchResults articles={mockArticles} />)
		expect(wrap).toMatchSnapshot()
    })
    
	test('returns the default empty array when there is no data', () => {
		const wrap = shallow(<SearchResults />)
		expect(wrap).toMatchSnapShot()
    })

})
