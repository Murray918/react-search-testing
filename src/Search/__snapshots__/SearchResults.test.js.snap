// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders the articles 1`] = `ShallowWrapper {}`;
